pages/deliverer/incoming-orders/[id].tsx

```diff
@@ -0,0 +1,201 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react'
+import { supabase } from '../../../api'
+import { useRouter } from 'next/router'
+import { useAlert } from '../../../helpers/alertContext'
+import { supabaseClient, withPageAuth } from '@supabase/auth-helpers-nextjs'
+
+type IncomingOrderDto = {
+  id: number
+  restaurants: RestaurantDto
+  order_text: string
+  deliverer_id: string
+  ordered_at: Date
+  is_active: boolean
+  destinations: DestinationDto
+  cost: number
+}
+
+type RestaurantDto = {
+  id: number
+  name: string
+}
+
+type DestinationDto = {
+  id: number
+  name: string
+}
+
+const IncomingOrder = ({ user }) => {
+  const router = useRouter()
+  const [alert, setAlert] = useAlert()
+
+  // will be passed in from the homescreen, when the user selects the incoming order
+  const { id } = router.query
+  const orderId: number = Number(id)
+
+  const [order, setOrder]: [
+    IncomingOrderDto,
+    Dispatch<SetStateAction<IncomingOrderDto>>,
+  ] = useState()
+
+  // simulate api call, pass in props.id
+  useEffect(() => {
+    const query = async () => {
+      const { data: ordersData, error: ordersError } = await supabaseClient
+        .from<IncomingOrderDto>('orders')
+        .select(
+          'id, restaurants (id, name), order_text, ordered_at, is_active, destinations (id, name), cost',
+        )
+        .eq('id', orderId)
+        .is('deliverer_id', null)
+        .limit(1)
+        .single()
+
+      if (ordersError) {
+        router.push('/home')
+        setAlert({
+          type: 'warning',
+          message: ordersError.code + ': ' + ordersError.message,
+        })
+        console.log(ordersError)
+        return
+      }
+
+      // means a deliverer has already takent the order
+      if (ordersData.deliverer_id) {
+        router.push('/home')
+        setAlert({ type: 'info', message: 'Order has been taken' })
+        return
+      }
+
+      console.log(ordersData)
+
+      setOrder({ ...ordersData, ordered_at: new Date(ordersData.ordered_at) })
+    }
+
+    query()
+
+    //not best practice, need to figure out how to handle this
+  }, [])
```

Comment:
this actually throws a warning, so we should figure out what to use as the dependency array

Reply:
added all the required dependencies, since none of them will change for a certain page